### PR TITLE
Postman's Datadog Integration, initial draft

### PR DESCRIPTION
**Pull request by shashankawasthi88**
### What does this PR do?

This PR is to add the Postman Integration to the Datadog platform

### Motivation

What inspired you to submit this pull request?
Postman in now listed as one of the technology partner with Datadog. This PR is in reference to the ongoing discussion between me from Postman and Alex Vetras from Datadog side.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?
Please don't make this PR live yet. Will need to verify this on the Sandbox first.

### CODE REVIEW

File: postman/metadata.csv [hunk 1]
```diff
@@ -0,0 +1,15 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
+PostmanMonitor_errors,gauge,,,,No of errors in an API run,0,Postman,
```
Comment (hithwen) [inline postman/metadata.csv:2]: Metrics usually follow the following convention: <integration_name>.<metric> and are all in lowercase

Comment (shashankawasthi88) [reply postman/metadata.csv:2]: Hi @hithwen thanks for pointing it out. Unfortunately, these metrics were implemented before we decided to create this integration. There are users who already have these metrics in production and hence renaming them might not be possible. 
Is it fine to keep this naming for the metrics that we have currently and follow the <integration_id>.<metric> for the new metric that we add.

File: postman/assets/dashboards/postman_overview.json [hunk 2]
```diff
@@ -0,0 +1,692 @@
+{
```
Comment (alexvetras) [inline postman/assets/dashboards/postman_overview.json:1]: Hi @shashankawasthi88 , can you please change this Dashboard "Timeboard" to a Dashboard "Screenboard" and include Postman's logo? 

Here are some dashboarding best practices with details on using screenboards: https://datadoghq.dev/integrations-core/guidelines/dashboards/.

Thanks!

**Comment by alexvetras** [comment]
Hi @shashankawasthi88, can you also add an SVG of Postman's logo to the "assets" folder? Our DesignOps team will use this to support your logo in the Datadog app, including light and dark modes.

**Comment by alexvetras** [comment]
Could you please review the failing checks, including the "PR Changes" and "PR Changes (Linux)"?

File: postman/assets/dashboards/postman_overview.json (hunk 2 above)
Comment (shashankawasthi88) [reply postman/assets/dashboards/postman_overview.json:1]: Alex, the latest commit has the Dashboard as "Screnboard" with a Logo. Can you verify?

Comment (alexvetras) [reply postman/assets/dashboards/postman_overview.json]: Hi Shashank, it seems your screenboard is not formatted correctly. When I copy the raw JSON from the current PR file and import it into a new screenboard in your account, I cannot get it to display. Can you please try to re-export the JSON in your screenboard and update the file? Thanks!

Comment (shashankawasthi88) [reply postman/assets/dashboards/postman_overview.json]: Updated, please check now.

**Comment by shashankawasthi88** [comment]
> Hi @shashankawasthi88, can you also add an SVG of Postman's logo to the "assets" folder? Our DesignOps team will use this to support your logo in the Datadog app, including light and dark modes.

Added SVG file, the same one works well for both modes in Postman.

**Comment by shashankawasthi88** [comment]
> Could you please review the failing checks, including the "PR Changes" and "PR Changes (Linux)"?
> 
> For context, it appears the CI is failing because Postman needs to add the integration to codeowners file https://github.com/DataDog/integrations-extras/blob/master/.github/CODEOWNERS

All tests passing now!

File: postman/metadata.csv (hunk 1 above)
Comment (hithwen) [reply postman/metadata.csv]: @alexvetras can you help here?

File: postman/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Agent Check: Postman
+
+## Overview
+
+[Postman][1] is a collaboration platform for API development. 
+Postman's features simplify each step of building an API and streamline collaboration so you can create better APIs—faster.
+This Integration helps you 
+1. Analyse the metric from the Postman Monitor runs in Datadog
```
Comment (jtappa) [inline postman/README.md:8]:
```suggestion
1. Analyze the metric from the Postman Monitor runs in Datadog
```

File: postman/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Agent Check: Postman
+
+## Overview
+
+[Postman][1] is a collaboration platform for API development. 
+Postman's features simplify each step of building an API and streamline collaboration so you can create better APIs—faster.
+This Integration helps you
```
Comment (jtappa) [inline postman/README.md:7]:
```suggestion
This Integration helps you:
```

File: postman/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Agent Check: Postman
+
+## Overview
+
+[Postman][1] is a collaboration platform for API development. 
+Postman's features simplify each step of building an API and streamline collaboration so you can create better APIs—faster.
+This Integration helps you 
+1. Analyse the metric from the Postman Monitor runs in Datadog
+2. Generate events on each monitor run and failure for you to remain on top of your monitor health
```
Comment (jtappa) [inline postman/README.md:9]:
```suggestion
2. Generate events on each monitor run and failure for you to remain on top of your monitor health

```

File: postman/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+## Setup
+
+For the detailed instructions, follow the [Postman documentation][3].
```
Comment (jtappa) [inline postman/README.md:12]:
```suggestion
For detailed instructions, follow the [Postman documentation][3].
```

File: postman/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+### Installation
+
+
+### Configuration
+
+1. Generate an API key from the Datadog Integrations> API tab
+2. Log into your Postman Account
```
Comment (jtappa) [inline postman/README.md:20]:
```suggestion
2. Log into your Postman account.
```

File: postman/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+## Setup
+
+For the detailed instructions, follow the [Postman documentation][3].
+
+### Installation
+
+
+### Configuration
+
+1. Generate an API key from the Datadog Integrations> API tab
```
Comment (jtappa) [inline postman/README.md:19]:
```suggestion
1. Generate an API key from the Datadog Integrations > API tab.
```

File: postman/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+2. Log into your Postman Account
+3. Go to Home > Integrations > Browse All Integrations
```
Comment (jtappa) [inline postman/README.md:21]:
```suggestion
3. Go to Home > Integrations > Browse All Integrations.
```

File: postman/README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+4. Click on the Datadog Integration
```
Comment (jtappa) [inline postman/README.md:22]:
```suggestion
4. Click on the Datadog Integration.
```

File: postman/README.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+5. Choose the Postman Monitor for which the metrics need to be analysed
```
Comment (jtappa) [inline postman/README.md:23]:
```suggestion
5. Choose the Postman Monitor for the metrics that you want to be analyzed.
```

File: postman/README.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+6. Choose if you want to send metric, events or both
```
Comment (jtappa) [inline postman/README.md:24]:
```suggestion
6. Choose if you want to send metrics, events or both.
```

File: postman/README.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+7. Complete other details and add the Integration to complete the configuration
```
Comment (jtappa) [inline postman/README.md:25]:
```suggestion
7. Complete other details and add the Integration to complete the configuration.
```

File: postman/README.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+See [metadata.csv][5] for a list of metrics provided by this integration.
+
+### Service Checks
+
+Postman does not include any service checks.
+
+### Events
+
+An event is generated each time a monitor runs in Postman. The severity of the event is Low if all the Tests in the Postman
```
Comment (jtappa) [inline postman/README.md:45]:
```suggestion
An event is generated each time a monitor runs in Postman. The severity of the event is "Low" if all the tests in the Postman 
```

File: postman/README.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+See [metadata.csv][5] for a list of metrics provided by this integration.
+
+### Service Checks
+
+Postman does not include any service checks.
+
+### Events
+
+An event is generated each time a monitor runs in Postman. The severity of the event is Low if all the Tests in the Postman 
+Monitor pass and Normal in case a few of them fail or if there is error in execution of any  event.
```
Comment (jtappa) [inline postman/README.md:46]:
```suggestion
Monitor pass and "Normal" if some of them fail, or if there is an error in execution of any event.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Comment by shashankawasthi88** [comment]
@jtappa I see that I have committed all the suggestions made by you, though I could see GitHub suggesting 1 change requested. Can you point me to the change that still needs to be made?

File: postman/manifest.json [hunk 16]
```diff
@@ -0,0 +1,34 @@
+{
+  "display_name": "Postman",
+  "maintainer": "integrations-partnerships@postman.com",
+  "manifest_version": "1.0.0",
+  "name": "postman",
+  "metric_prefix": "PostmanMonitor_",
+  "metric_to_check": "",
```
Comment (mgarabed) [inline postman/manifest.json:7]: This should be populated with a metric expected to always be reported to determine if the check is working correctly.

**Review by mgarabed — CHANGES_REQUESTED** [review]

**Review by jtappa — APPROVED** [review]

File: postman/manifest.json [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  "creates_events": true,
+  "short_description": "Integration to observe metric from Postman's Monitor run in Datadog",
```
Comment (alexvetras) [inline postman/manifest.json:9]: metric --> metrics

I also think you can delete the word "run"

Comment (shashankawasthi88) [reply postman/manifest.json]: Updated

File: postman/manifest.json (hunk 16 above)
Comment (shashankawasthi88) [reply postman/manifest.json]: Updated. Also updated the metric_prefix according to the new naming convention.

File: postman/metadata.csv [hunk 18]
```diff
@@ -0,0 +1,15 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
+postman.monitor.run.errors,gauge,,,,No of errors in an API run,0,postman,
+Ppostman.monitor.run.failed_tests,gauge,,,,No of failed tests in an API run,0,postman,
```
Comment (mgarabed) [inline postman/metadata.csv:3]:
```suggestion
postman.monitor.run.failed_tests,gauge,,,,No of failed tests in an API run,0,postman,
```

**Review by mgarabed — CHANGES_REQUESTED** [review]

**Comment by shashankawasthi88** [comment]
@alexvetras @mgarabed I see that the logo used in other dashboard have a static path /static/images/logos..... I wanted to see if we can do that with our dashboard as well. In which case where do we actually check in this image?

**Comment by mgarabed** [comment]
@shashankawasthi88 Just opened a ticket internally to get a logo mounted to our static assets.  Once that task completes the dashboard file can be updated with a reference to the image.

**Comment by mgarabed** [comment]
@shashankawasthi88 logos have been uploaded, feel free to choose one of the following URLs:
- `/static/images/logos/postman_avatar.svg`
- `/static/images/logos/postman_small.svg`
- `/static/images/logos/postman_large.svg`

Example: https://github.com/DataDog/integrations-extras/blob/8c40a9cf870578687cc224ee91d3c70cd3a435a4/speedtest/assets/dashboards/speedtest.json#L139

**Comment by shashankawasthi88** [comment]
> @shashankawasthi88 logos have been uploaded, feel free to choose one of the following URLs:
> 
> * `/static/images/logos/postman_avatar.svg`
> * `/static/images/logos/postman_small.svg`
> * `/static/images/logos/postman_large.svg`
> 
> Example:
> 
> https://github.com/DataDog/integrations-extras/blob/8c40a9cf870578687cc224ee91d3c70cd3a435a4/speedtest/assets/dashboards/speedtest.json#L139

@mgarabed thanks for the help, is there a way I could visualize how this would look? I tried installing the Integration on the sandbox environment to see the default dashboard, but the installation gets hangged.

File: postman/README.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+For detailed instructions, follow the [Postman documentation][3].
+
+
+
+
+### Configuration
+
+1. Generate a Datadog [API key][6].
+2. Sign in to your Postman account and navigate to the [Datadog integration][7].
+3. Select “Add Integration.”
```
Comment (mgarabed) [inline postman/README.md:24]:
```suggestion
3. Select "Add Integration."
```

File: postman/README.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+4. To send your monitor metrics and events to Datadog:
+- Name your new integration.
+- Select the monitor whose data you would like to send to Datadog.
+- Enter your Datadog API key.
+- Select the Datadog region you would like to use.
+- Optionally choose if you want to send events, metrics or both for each run.
+5. Then select “Add Integration” to finish setting up the integration.
```
Comment (mgarabed) [inline postman/README.md:31]:
```suggestion
5. Then select "Add Integration" to finish setting up the integration.
```

File: postman/README.md [hunk 21]
```diff
@@ -0,0 +1,64 @@
+# Agent Check: Postman
+
+## Overview
+
+[Postman][1] is a collaboration platform for API development. It simplifies the steps of building an API and streamlines 
+collaboration so you can create better APIs—faster.
```
Comment (mgarabed) [inline postman/README.md:6]:
```suggestion
collaboration so you can create better APIs-faster.
```

**Review by mgarabed — CHANGES_REQUESTED** [review]
These non-ascii characters cause parsing errors in the backend system.

**Comment by shashankawasthi88** [comment]
@alexvetras the build seems to be failing due to other project's manifest files. 
Also the I renamed a few metrics, and it seems on the sandbox environment both the old and the new metric are present. Is there something additional that I need to do here ?

**Comment by mgarabed** [comment]
> @alexvetras the build seems to be failing due to other project's manifest files.

@shashankawasthi88 We added a new validation to our scripts, but no changes are needed for this integration (will be fixed separately).

The other issue is under investigation.

**Comment by mgarabed** [comment]
> Also the I renamed a few metrics, and it seems on the sandbox environment both the old and the new metric are present. Is there something additional that I need to do here ?

@shashankawasthi88 can you incorporate the minor text changes above to the README?  Once those are merged, I will re-sync.

**Comment by shashankawasthi88** [comment]
> > Also the I renamed a few metrics, and it seems on the sandbox environment both the old and the new metric are present. Is there something additional that I need to do here ?
> 
> @shashankawasthi88 can you incorporate the minor text changes above to the README? Once those are merged, I will re-sync.

@mgarabed - incorporated.

**Comment by shashankawasthi88** [comment]
@mgarabed the Integration on our Datadog environment is hanging a lot from time to time. We need to demo this stuff to the stakeholders tomorrow. I believe it's because of the characters that you pointed out above. Let me know if there are any other errors that need to be corrected


https://user-images.githubusercontent.com/71059054/113721823-7ba96480-970d-11eb-9299-d57f96cdbb73.mov

**Comment by mgarabed** [comment]
@shashankawasthi88 The hanging issue should be resolved based on a few tests - can you try again and report if you are still seeing it?

The duplicate metadata updates will be fixed today.

**Comment by mgarabed** [comment]
@shashankawasthi88 Metadata should be cleaned up now:

Screen Shot 2021-04-06 at 3 37 57 PM

**Comment by shashankawasthi88** [comment]
@mgarabed the hanging issues seem to be resolved now. Also the meta data is cleaned. I have made small updates in the Dashboard content. On installing the Integration, I don't see the dashboard being listed though? I wanted to test how the Logo looks as I am using the static image URL now. I could not see the dashboard being created. Let me know what is needed for that to appear.

**Comment by mgarabed** [comment]
@shashankawasthi88 Dashboard synced.

**Comment by shashankawasthi88** [comment]
@mgarabed @AlexandreYang I have pushed final changes from my end. Made minor wording changes. Feel free to have a go at it.
There is one thing that I noticed though, the markdown bullets are not working fine in here 
Screenshot 2021-04-13 at 1 22 33 PM
My markdown editor shows this bullet point correctly in continuation with the points above, though I could see that's not how it appears here on the UI.

File: postman/README.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+1. Generate a Datadog [API key][6].
+2. Sign in to your Postman account and navigate to the [Datadog integration][7].
+3. Select "Add Integration."
+4. To send your monitor metrics and events to Datadog:
+- Name your new integration.
+- Select the monitor whose data you would like to send to Datadog.
+- Enter your Datadog API key.
+- Select the Datadog region you would like to use.
+- Optionally choose if you want to send events, metrics or both for each run.
```
Comment (jtappa) [inline postman/README.md:30]:
```suggestion
   - Name your new integration.
   - Select the monitor whose data you would like to send to Datadog.
   - Enter your Datadog API key.
   - Select the Datadog region you would like to use.
   - Optionally choose if you want to send events, metrics or both for each run.
```

**Comment by jtappa** [comment]
Sorry I missed the indentation issue when I reviewed this! Whenever you want a child in an ordered list in markdown, the first character has to be in the same position as the first character of it's parent. So in this cases, these just needed 3 spaces before each and they should render properly inside of #4. Hope this helps!

**Review by hithwen — APPROVED** [review]

**Comment by shashankawasthi88** [comment]
@alexvetras @hithwen is this merged to Production? As discussed earlier, we want to wait until we make changes (rename metrics) at our end. I hope this is not live on production.

**Comment by hithwen** [comment]
Ohh, I'm very sorry, because you said

> I have pushed final changes from my end

I assumed you were done and had indeed published the tile.
Because `is_public` is set to false on the `manifest` this wont show on the docs page (you'll need to open a PR to fix that once you are done). I've now removed the tile from the integrations page

**Pull request merged**